modules/node/handler/node_form_response.e

```diff
@@ -302,6 +359,45 @@ feature -- Form
 			populate_form (a_node_type, f, a_node)
 			f.extend_html_text ("<br/>")
 
+				-- Select for nodes
+			create tl.make ("select_parent_node")
+			create to.make ("0", "NONE")
+			tl.add_option (to)
+
+			if attached {CMS_PAGE} a_node as l_node_page then
+				if attached l_node_page.parent as l_parent_node then
+					f.extend_html_text ("<div><strong>Parent page is </strong> ")
+					f.extend_html_text (node_html_link (l_parent_node,  l_parent_node.title + "(#" + l_parent_node.id.out + ")"))
+					f.extend_html_text ("<br/>")
+					f.extend_html_text ("Change parent to?")
+					if a_node /= Void then
+						across node_api.available_parents_for_node (a_node) as ic loop
```

Comment:
Yes, we can do that, but with available parents we are sure we show all potentially valid parents, if we let the user add the id, then we need to validate it, for example, a node can not add as parent itself, etc.

In the end ask only for the parent id is more flexible.